src/open_producten/producttypen/serializers/producttype.py

```diff
@@ -0,0 +1,160 @@
+# from open_producten.locaties.serializers.location import (
+#     ContactSerializer,
+#     LocationSerializer,
+#     OrganisationSerializer,
+# )
+from open_producten.utils.serializers import build_array_duplicates_error_message
+
+from ..models import Onderwerp, ProductType, UniformeProductNaam
+from .bestand import BestandSerializer
+from .link import LinkSerializer
+from .prijs import PrijsSerializer
+from .vraag import VraagSerializer
+
+
+class SimpleOnderwerpSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Onderwerp
+        exclude = ("path", "depth", "numchild")
+
+
+class ProductTypeSerializer(serializers.ModelSerializer):
+    uniforme_product_naam = serializers.SlugRelatedField(
+        slug_field="uri", queryset=UniformeProductNaam.objects.all()
+    )
+
+    onderwerpen = SimpleOnderwerpSerializer(many=True, read_only=True)
+    onderwerp_ids = serializers.PrimaryKeyRelatedField(
+        many=True,
+        write_only=True,
+        queryset=Onderwerp.objects.all(),
+        source="onderwerpen",
+    )
+
+    # locaties = LocationSerializer(many=True, read_only=True)
+    # locatie_ids = serializers.PrimaryKeyRelatedField(
+    #     many=True,
+    #     write_only=True,
+    #     queryset=Locatie.objects.all(),
+    #     default=[],
+    #     source="locaties",
+    # )
+    #
+    # organisaties = OrganisationSerializer(many=True, read_only=True)
+    # organisatie_ids = serializers.PrimaryKeyRelatedField(
+    #     many=True,
+    #     write_only=True,
+    #     queryset=Organisatie.objects.all(),
+    #     default=[],
+    #     source="organisaties",
+    # )
+    #
+    # contacts = ContactSerializer(many=True, read_only=True)
+    # contact_ids = serializers.PrimaryKeyRelatedField(
+    #     many=True,
+    #     write_only=True,
+    #     queryset=Contact.objects.all(),
+    #     default=[],
+    #     source="contacten",
+    # )
+
+    vragen = VraagSerializer(many=True, read_only=True)
+    prijzen = PrijsSerializer(many=True, read_only=True)
+    links = LinkSerializer(many=True, read_only=True)
+    bestanden = BestandSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = ProductType
+        fields = "__all__"
+
+    def validate_onderwerp_ids(self, onderwerpen: list[Onderwerp]) -> list[Onderwerp]:
+        if len(onderwerpen) == 0:
+            raise serializers.ValidationError(
+                _("Er is minimaal één onderwerp vereist.")
+            )
+        return onderwerpen
+
+    def _handle_relations(
+        self,
+        *,
+        instance,
+        onderwerpen: list[Onderwerp],
+        # locaties: list[Locatie],
+        # organisaties: list[Organisatie],
+        # contacten: list[Contact],
+    ) -> None:
+        errors = dict()
+        if onderwerpen is not None:
+            build_array_duplicates_error_message(onderwerpen, "onderwerp_ids", errors)
+            instance.onderwerpen.set(onderwerpen)
```

Comment:
Same comment as oderwerpen. Put the set() after if errors and put if errors inside the `if onderwerpen is not None:`